src/server/dfly_main.cc

```diff
@@ -161,9 +162,33 @@ struct VersionMonitor {
   }
 
  private:
+  bool IsVersionOutdated(std::string_view remote, std::string_view current) const;
   void RunTask(SSL_CTX* ssl_ctx);
 };
 
+bool VersionMonitor::IsVersionOutdated(const std::string_view remote,
+                                       const std::string_view current) const {
+  const std::vector<absl::string_view> remote_xyz = absl::StrSplit(remote, ".");
+  const std::vector<absl::string_view> current_xyz = absl::StrSplit(current, ".");
+  if (remote_xyz.size() != current_xyz.size()) {
+    LOG(WARNING) << "Remote and current version fields must be of equal size";
+    return false;
+  }
+  for (size_t i = 0; i < remote_xyz.size(); ++i) {
+    size_t remote_x = 0;
+    size_t current_x = 0;
+    if (!absl::SimpleAtoi(remote, &remote_x) || !absl::SimpleAtoi(current, &current_x)) {
+      LOG(WARNING) << "Conversion error on version string_view";
```

Comment:
Same here, especially with "string_view" which will confuse readers I think :)
Maybe `Can't parse " << ... << " part of version " << remote << " as a number";` ?